src/providers/oauth-provider.js

```diff
@@ -85,10 +89,21 @@ function OAuthProvider() {
        */
 
       constructor() {
-        if (!config) {
+        if (!defaultConfig) {
```

Comment:
Another closure is created here. To improve this I would recommend to pass `defaultConfig`so you can assign this to a property of `OAuthService`.

```js
class OAuth {
  constructor(config) {
    this.config = config;
  }

  configure(params) {
    this.config = sanitizeConfigParams(params);
  }

  // ...
}

return new OAuth(this.defaultConfig);
```